### PR TITLE
Adjust semaphore test timeouts

### LINKED ISSUE #3717: Test failure in test_oversubscribing_leases
This showed up in a recent PR.  I copy-pasted the traceback and then restarted the job. 
 Unfortunately my attempt to copy-paste the trace turned out to be incomplete.  My apologies.  Hopefully this helps though.

cc @fjetter 

```python-traceback
>   assert not sem.acquire(timeout=0.05)
E   assert not True
E    +  where True = <bound method Semaphore.acquire of <distributed.semaphore.Semaphore object at 0x0000027BF23DE438>>(timeout=0.05)
E    +    where <bound method Semaphore.acquire of <distributed.semaphore.Semaphore object at 0x0000027BF23DE438>> = <distributed.semaphore.Semaphore object at 0x0000027BF23DE438>.acquire

distributed\tests\test_semaphore.py:396: AssertionError
---------------------------- Captured stderr call -----------------------------
distributed.worker - WARNING -  Compute Failed
Function:  observe_state
args:      ()
kwargs:    {'sem': <distributed.semaphore.Semaphore object at 0x0000027BF23DE438>}
Exception: AssertionError('assert not True\n +  where True = <bound method Semaphore.acquire of <distributed.semaphore.Semaphore object at 0x0000027BF23DE438>>(timeout=0.05)\n +    where <bound method Semaphore.acquire of <distributed.semaphore.Semaphore object at 0x0000027BF23DE438>> = <distributed.semaphore.Semaphore object at 0x0000027BF23DE438>.acquire',)

```

### PR DESCRIPTION
**Pull request by fjetter**
There is a race condition between "refresh started again" and "observer tries to acquire". With an additional wait this should not happen anymore.

Closes #3717

### CODE REVIEW

**Comment by fjetter** [comment]
This is the only place I can see a race condition happening but I couldn't reproduce it locally since it is a quite narrow window. I'll try to play with the settings to confirm my suspicion but I'm fairly confident this should do the trick

cc @mrocklin @lr4d

File: distributed/tests/test_semaphore.py [hunk 1]
```diff
@@ -361,7 +361,8 @@ def guaranteed_lease_timeout(x, sem):
 
         with sem:
             # This simulates a task which holds the GIL for longer than the
-            # lease-timeout.
+            # lease-timeout. This is twice the lease timeout to ensurre that the
```
Comment (lr4d) [inline distributed/tests/test_semaphore.py:364]:
```suggestion
            # lease-timeout. This is twice the lease timeout to ensure that the
```

File: distributed/tests/test_semaphore.py [hunk 2]
```diff
@@ -391,9 +392,12 @@ def observe_state(sem):
             x_locked = client.get_metadata(0) == "locked"
             y_locked = client.get_metadata(1) == "locked"
 
+        # Once both are locked we should give the refresh time to notify the scheduler
+        # This parameter should be larger than ``lease-validation-interval``
+        slowidentity(delay=0.15)
```
Comment (lr4d) [inline distributed/tests/test_semaphore.py:397]: lease-validation-interval is 10ms, do you mean `0.015`?

Comment (fjetter) [reply distributed/tests/test_semaphore.py:397]: well, anything larger should do

**Comment by fjetter** [comment]
Will merge and will not include the above mentioned nitpicks in the interest of not blocking anybody while waiting for another CI pass

**Pull request merged**